insights/parsers/mongod_conf.py

```diff
@@ -0,0 +1,197 @@
+
+    # Where and how to store data.
+    storage:
+      dbPath: /var/lib/mongo
+      journal:
+        enabled: true
+
+
+Sample input(key-value pair format)::
+
+    # mongodb.conf - generated from Puppet
+    #where to log
+    logpath=/var/log/mongodb/mongodb.log
+    logappend=true
+    # Set this option to configure the mongod or mongos process to bind to and
+    # listen for connections from applications on this address.
+    # You may concatenate a list of comma separated values to bind mongod to multiple IP addresses.
+    bind_ip = 127.0.0.1
+    # fork and run in background
+    fork=true
+    dbpath=/var/lib/mongodb
+    # location of pidfile
+    pidfilepath=/var/run/mongodb/mongodb.pid
+    # Enables journaling
+    journal = true
+    # Turn on/off security.  Off is currently the default
+    noauth=true
+
+
+Examples:
+
+    >>> mongod_conf1 = shared[MongodConf]
+    >>> mongod_conf2 = shared[MongodConf]
+    >>> MongodbConf1.is_yaml
+    True
+    >>> MongodbConf2.is_yaml
+    False
+    >>> mongod_conf1.fork
+    True
+    >>> mongod_conf2.fork
+    'true'
+    >>> mongod_conf1.dbpath
+    '/var/lib/mongo'
+    >>> mongod_conf2.dbpath
+    '/var/lib/mongo'
+    >>> mongod_conf1.get("systemlog", {}).get("logAppend")
+    True
+    >>> MongodbConf2.get("logappend")
+    'true'
+
+"""
+
+import yaml
+from .. import parser, Parser, LegacyItemAccess, get_active_lines
+from ..parsers import ParseException, split_kv_pairs
+
+
+@parser('mongod_conf')
+class MongodbConf(Parser, LegacyItemAccess):
+    """
+    Parse config file in key-value pair or YAML format.
+    Make several frequently used config options as properties.
+    Attribute 'is_yaml' is a boolean value, flags if this is a yaml format file.
+
+    Raises:
+        ParseException: Raised when any problem parsing the file content.
+    """
+    def parse_content(self, content):
+        self.is_yaml = self._file_type_is_yaml(content)
+        try:
+            if self.is_yaml:
+                self.data = yaml.safe_load('\n'.join(content))
+            else:
+                self.data = split_kv_pairs(content, use_partition=True)
+        except Exception, e:
+            raise ParseException('mongod conf parse failed: %s', e)
+
+    def _file_type_is_yaml(self, content):
+        """
+        Return True if the file type is YAML.
+        Return False means this file will be handled in key-value pair format.
+
+        Why 0.9?
+            The normal key-value pair format file would always has the '='
+            in each line. Use 0.9 rather than 1 here, just in case there're
+            any unexpected lines with wrong settings.
+        """
+        content = get_active_lines(content)
+        cnt = sum([1 for line in content if "=" in line])
+        percent = float(cnt) / len(content)
+        return True if percent < 0.9 else False
+
+    @property
+    def bindip(self):
+        """
+        Return option value of:
+            'net.bindIp' in yaml conf;
```

Comment:
It might be clearer if you use backticks to highlight these values. For instance `net.bindIp`.  Also since this renders as just a single line, you might want to simply use prose to describe it, for instance:
```
Returns option value of `net.bindIp` if a yaml conf and `bind_ip` if a key-value pair conf.
```